File: location-cloud/src/main/kotlin/nl/cloud/location/domain/path/TimedGeoLocation.kt
package nl.cloud.location.domain.path

/**
 * Represents a location and a moment in time.
 */
data class TimedGeoLocation(
    val longitude: Double,
    val latitude: Double,
    val timestamp: Long
) {
    init {
        if( longitude < -180 || longitude > 180 ) {
            throw IllegalArgumentException("longitude out of range")
        } else if( latitude < -90 || latitude > 90 ) {
            throw IllegalArgumentException("latitude out of range")
        } else if( timestamp < 0 ) {
            throw IllegalArgumentException("timestamp less than 0")
        }
    }

    override fun toString(): String {
        return "TimedGeoLocation(longitude=$longitude, latitude=$latitude, timestamp=$timestamp)"
    }

}

File: location-ui/src/domain/TimedGeoLocation.js
import GoogleGeoLocation from './GoogleGeoLocation'

/**
* Represents a location on earth at a certain time.
**/
class TimedGeoLocation {

    constructor(timestamp, latitude, longitude) {
        this.timestamp = timestamp;
		this.latitude = latitude;
		this.longitude = longitude;
    }

	toGoogleGeoLocation = () => {
		return new GoogleGeoLocation(this.latitude, this.longitude);
	}

}

export default TimedGeoLocation
